superbench/benchmarks/micro_benchmarks/cpu_stream_performance.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module for running the University of Virginia STREAM tool. It measures sustainable main memory
+    bandwidth in MB/s and the corresponding computation rate for simple vector kernels."""
+
+import os
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CpuStreamBenchmark(MicroBenchmarkWithInvoke):
+    """The Stream benchmark class."""
+
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'streamZen3.exe'
+        self.__cpu_arch = ['other', 'zen3', 'zen4']
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--CPU_ARCH',
+            type=str,
+            default=['other'],
+            required=False,
+            help='The targeted cpu architectures to run \
+                STREAM. Possible values are {}.'.format(' '.join(self.__cpu_arch))
+        )
+        self._parser.add_argument(
+            '--cores',
+            nargs='+',
+            type=int,
+            required=True,
+            help='List of cores to perform test'
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+        if not self._set_binary_path():
+            return False
+
+        # zen3
+        # cores=[0, 4, 8, 12, 16, 20, 24, 28, 30, 34, 38, 42, 46, 50,
+        # 54, 58, 60, 64, 68, 72, 76, 80, 84, 88, 90, 94, 98, 102, 106, 110, 114, 118]
```

Comment:
Yea, this will be passed in via config file so it is ultimately up to the end user. However, 1 core is chosen per ccx. The reason for the non-uniformity is due to the number of cores that are held back for min root. For some HB series you can view which cores belong to which L3 cache and use that (its not always the case due to how physical and virtual core mapping). These cores used in customer facing msft documentation of STREAM on HBv3.

Reply:
maybe add the core config document link in "--cores" argument help message?

Reply:
Yea, it seems that the cores chosen for HBv4 were deduced from the P core to V core mapping. I don't think there is a way to expose that to a customer/user. We can definitely add a link to the public facing doc for HBv3. However, this will only cover HBv3. The HBv4 stream core configuration was created by our team. Here is the link for [HBv3](https://techcommunity.microsoft.com/t5/azure-compute-blog/performance-amp-scalability-of-hbv3-vms-with-milan-x-cpus/ba-p/2939814).  With that said I can attempt to add more info in the help message.

Reply:
I'm still working on this please bare with me

Reply:
addressed